src/app/viewer/news/news.component.ts

```diff
@@ -1,15 +1,36 @@
 import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router, ParamMap } from '@angular/router';
+import { NewsService } from 'src/app/shared/services/news.service';
+import { mergeMap } from 'rxjs/operators';
 
 @Component({
   selector: 'app-news',
   templateUrl: './news.component.html',
   styleUrls: ['./news.component.scss']
 })
 export class NewsComponent implements OnInit {
+  private id: number;
+  private mediaPath: string;
+  private mediaCaption: string;
+  private newsItem: any;
 
-  constructor() { }
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private newsService: NewsService,
+  ) { }
 
   ngOnInit() {
+    this.id = parseInt(this.route.snapshot.paramMap.get('id'), 10);
+    this.newsService.getNewsItem(this.id)
+    .pipe(mergeMap(data => {
+      this.newsItem = data;
+      return this.newsService.getNewsMedia(this.newsItem.featured_media);
+    }))
```

Comment:
Something for mulling, but there's a balance point to be struck between having granular data calls which can be reused vs larger calls which produce View Models to reduce the amount of logic/knowledge needed in components.

This might come up through the Architecture Runway, where it seems like some of the responsibility of the API Services/Gateway would be to provide a kind of backend-for-frontend.